src/sonic-yang-models/yang-models/sonic-system-port.yang

```diff
@@ -0,0 +1,81 @@
+module sonic-system-port {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-system-port";
+    prefix system-port;
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "SYSTEM_PORT YANG Module for SONiC OS";
+
+    revision 2022-11-10 {
+        description "First Revision";
+    }
+
+    container sonic-system-port {
+
+        container SYSTEM_PORT {
+
+            description "SYSTEM_PORT table of config_db.json";
+
+            list SYSTEM_PORT_LIST {
+
+                key "hostname asic_name ifname";
+
+                leaf hostname {
+                    type stypes:hostname;
+                }
+
+                leaf asic_name {
+                    type string;
+                }
+
+                leaf ifname {
+                    type string;
+                }
+
+                leaf core_index {
+                    type uint8;
+                    description "Hardware core index a port is affiliated with.";
+                }
+
+                leaf core_port_index {
+                    type uint16;
```

Comment:
Allowed 0-2^16-1 ?

Reply:
Yes, similar to core_index we would like flexibility to support logical indices that can exceed 8 bits.